### PR TITLE
feat(map): port lanelet2_map_preprocessor from autoware.universe

### PR DESCRIPTION
**Pull request by a-maumau**
## Description
This PR will add `map/autoware_lanelet2_map_utils` which is a port of [`autoware.universe/map/util/lanelet2_map_preprocessor`](https://github.com/autowarefoundation/autoware.universe/tree/main/map/util/lanelet2_map_preprocessor) .

After this PR, the `lanelet2_map_preprocessor` in the autoware.universe  will be get removed and only the link to this package remains.

Changes
- add `autoware_lanelet2_map_utils`
- add dependency `autoware_lanelet2_extension` to `build_depends.repo`

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: map/autoware_lanelet2_map_utils/package.xml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+  <name>autoware_lanelet2_map_utils</name>
+  <version>0.1.0</version>
+  <description>The autoware_lanelet2_map_utils package</description>
+  <maintainer email="yamato.ando@tier4.jp">Yamato Ando</maintainer>
+  <maintainer email="kento.yabuuchi.2@tier4.jp">Kento Yabuuchi</maintainer>
```
Comment (YamatoAndo) [inline map/autoware_lanelet2_map_utils/package.xml:8]: please remove
```suggestion
```

File: map/autoware_lanelet2_map_utils/src/fix_lane_change_tags.cpp [hunk 2]
```diff
@@ -0,0 +1,104 @@
+// Copyright 2020 Tier IV, Inc.
```
Comment (YamatoAndo) [inline map/autoware_lanelet2_map_utils/src/fix_lane_change_tags.cpp:1]:
```suggestion
// Copyright 2020 TIER IV, Inc.
```

**Comment by a-maumau** [comment]
@YamatoAndo I applied your suggestions in 7b44197 and 2d22105 .

**Comment by a-maumau** [comment]
I change the `tier` to `TIER` for other files too.

**Review by YamatoAndo — APPROVED** [review]

**Pull request merged**